compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1274,11 +1476,51 @@ private BLangType createTypeNode(Node type) {
             bLUserDefinedType.pos = getPosition(type);
 
             return bLUserDefinedType;
+        } else if (type.kind() == SyntaxKind.INDEXED_EXPRESSION) {
+            return createBLangArrayFromIndexedExpressionNode((IndexedExpressionNode) type);
         } else {
             return (BLangType) type.apply(this);
         }
     }
 
+    private BLangArrayType createBLangArrayFromIndexedExpressionNode(IndexedExpressionNode indexedExpressionNode) {
+        int dimensions = 1;
+        List<Integer> sizes = new ArrayList<>();
+        do {
+            if (indexedExpressionNode.keyExpression() == null) {
+                sizes.add(UNSEALED_ARRAY_INDICATOR);
+            } else {
+                if (indexedExpressionNode.keyExpression().kind() == SyntaxKind.DECIMAL_INTEGER_LITERAL) {
+                    sizes.add(Integer.parseInt(indexedExpressionNode.keyExpression().toString()));
+                } else if (indexedExpressionNode.keyExpression().kind() == SyntaxKind.ASTERISK_TOKEN) {
+                    sizes.add(OPEN_SEALED_ARRAY_INDICATOR);
+                } else {
+                    // TODO : should handle the const-reference-expr
+                }
+            }
+            if (indexedExpressionNode.containerExpression().kind() != SyntaxKind.INDEXED_EXPRESSION) {
+                break;
+            }
+            indexedExpressionNode = (IndexedExpressionNode) indexedExpressionNode.containerExpression();
+            dimensions++;
+        } while (true);
```

Comment:
Can't use a while loop?

Reply:
I used do-while use because if iteration should be executed.

Reply:
But you are not terminating based on a condition in the `while()`. So it anyway runs the first one right?